### PR TITLE
feat(FX-4657): mobile web search overlay

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Feat**

This PR solves [FX-4657]

### Description

- create a new full screen overlay/modal 
- if feature flag is enabled, when user presses the search input (on mWeb only) they see the new overlay
- overlay should have an x button to dismiss
- overlay should include search input as specified in design

### Demo

https://github.com/artsy/force/assets/3934579/1d11f0c4-7982-4c95-ac08-b01677ac36a1

[FX-4657]: https://artsyproduct.atlassian.net/browse/FX-4657?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Search/NewSearch/NewSearchBarInput.tsx [hunk 1]
```diff
@@ -172,14 +163,9 @@ export const NewSearchBarInputQueryRenderer: FC<NewSearchBarInputQueryRendererPr
...
-      render={({ props: relayProps }) => {
-        if (relayProps && relayProps.viewer) {
-          return (
-            <NewSearchBarInputRefetchContainer
-              viewer={relayProps.viewer}
-              isXs={props.isXs}
-            />
-          )
+      render={({ props }) => {
+        if (props && props.viewer) {
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/NewSearchBarInput.tsx:167]: **nitpick**: `props?.viewer`

File: src/Components/Search/NewSearch/Mobile/SearchBar.tsx
Comment (dimatretyak) [inline src/Components/Search/NewSearch/Mobile/SearchBar.tsx:1]: **question**: IMHO we don't need to add and use `Mobile` folder here

File: src/Components/Search/NewSearch/Mobile/OverlayBase.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        width: "100%",
+        height: "100%",
+        background: "black",
+        justifyContent: "center",
+        backgroundColor: "white",
+      }}
+    >
+      <ModalDialogContent
+        onClose={onClose}
+        width={"100%"}
```
Comment (dimatretyak) [inline src/Components/Search/NewSearch/Mobile/OverlayBase.tsx:21]: **nitpick**: `width="100%"` and `height="100%"`

**Review by dimatretyak — APPROVED** [review]
LGTM 👍

File: src/Components/Search/NewSearch/Mobile/SearchBar.tsx
Comment (nickskalkin) [reply src/Components/Search/NewSearch/Mobile/SearchBar.tsx:1]: I expect there to be a couple more files, all of them will be related only to a mWeb view. I'm not strongly opinionated, but I want to separate somehow components with similar names between Desktop / mWeb.

Comment (nickskalkin) [reply src/Components/Search/NewSearch/Mobile/SearchBar.tsx:1]: I'll merge like it is now, and refactor later if you don't mind (there will be ~ 3 or more PRs with related changes)

File: src/Components/Search/NewSearch/Mobile/OverlayBase.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export const OverlayBase: FC<OverlayBaseProps> = ({ children, onClose }) => {
+  return (
+    <ModalBase
+      dialogProps={{
+        width: "100%",
+        height: "100%",
+        background: "black",
+        justifyContent: "center",
+        backgroundColor: "white",
```
Comment (dzucconi) [inline src/Components/Search/NewSearch/Mobile/OverlayBase.tsx:16]: `white100`

File: src/Components/Search/NewSearch/Mobile/OverlayBase.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  onClose: () => void
+}
+
+export const OverlayBase: FC<OverlayBaseProps> = ({ children, onClose }) => {
+  return (
+    <ModalBase
+      dialogProps={{
+        width: "100%",
+        height: "100%",
+        background: "black",
```
Comment (dzucconi) [inline src/Components/Search/NewSearch/Mobile/OverlayBase.tsx:14]: `black100`

File: src/Components/Search/NewSearch/Mobile/OverlayBase.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        justifyContent: "center",
+        backgroundColor: "white",
+      }}
+    >
+      <ModalDialogContent
+        onClose={onClose}
+        width="100%"
+        height="100%"
+        margin={0} // TODO: ignored, fix it in palette
```
Comment (dzucconi) [inline src/Components/Search/NewSearch/Mobile/OverlayBase.tsx:23]: If we're altering the dialog then just don't use it

Comment (dzucconi) [reply src/Components/Search/NewSearch/Mobile/OverlayBase.tsx:23]: (Though, you're right, this should be fixed or the type should be altered)

**Pull request merged**